modules/layers/src/bitmap-layer/bitmap-layer.js

```diff
@@ -180,9 +179,34 @@ export default class BitmapLayer extends Layer {
     );
   }
 
-  draw({uniforms}) {
+  draw(opts) {
+    const {uniforms} = opts;
     const {bitmapTexture, model} = this.state;
-    const {desaturate, transparentColor, tintColor} = this.props;
+    const {image, desaturate, transparentColor, tintColor} = this.props;
+
+    // Update video frame
+    if (
+      bitmapTexture &&
+      image instanceof HTMLVideoElement &&
+      image.readyState > HTMLVideoElement.HAVE_METADATA
+    ) {
+      const sizeChanged =
+        bitmapTexture.width !== image.videoWidth || bitmapTexture.height !== image.videoHeight;
+      if (sizeChanged) {
+        // note clears image and mipmaps when resizing
+        bitmapTexture.resize({width: image.videoWidth, height: image.videoHeight, mipmaps: true});
+        bitmapTexture.setImageData({
+          data: image,
+          paramters: DEFAULT_TEXTURE_PARAMETERS
+        });
+      } else {
+        bitmapTexture.setImageData({
```

Comment:
subImageData

Reply:
what is the difference between `subImageData` and `setImageData`?

Reply:
`subImageData` skips some allocation operations and upload the data directly.